### ISSUE TITLE
[FEA] Audit enum fields re-exported from cudart to see if we can use the symbols (rather than matching values)

### ISSUE THREAD

**Issue by wence-** [rapidsai/rmm#1745]
**Is your feature request related to a problem? Please describe.**

See discussion in https://github.com/rapidsai/rmm/pull/1743, we have an `allocation_handle_type` enum that re-exports enum values from cudart. In the past we copied values because the symbols didn't appear in cudart 11.2. It may be that they now exist in all supported versions.

This would mean that we don't accidentally end up with mismatching values between cudart and rmm. One hopes that the enum values are fowards compatible forever, but ...

**Comment by bdice** [comment]
Yes, this is a good idea. For any enum values that are now present in all supported CUDA versions, we can do this. That would mean CUDA 11.2+ for sure, and possibly 11.4+. (We currently have some inconsistency in RMM around what our minimum CUDA version really is.)

**Comment by bdice** [comment]
I improved this in #1984 while I was touching some related documentation. I think the changes I made there should be enough to close this issue.
